### DISCUSSION TITLE
OTBR Leader election takes more time

### DISCUSSION THREAD

**Discussion by muralidhar-bn** [openthread/openthread#10669] [Q&A]
I have setup Raspberry Pi 4 - 2 devices with OTBR agent - One becomes a leader and other as router. Along with this 3 nanoleaf bulbs are joined to Thread network to form a mesh network. When i try to switch of Leader OTBR i observed electing new Leader takes lot of time 3 ~ 10 minutes. Is this normal behaviour ?

**Comment by jwhui** [comment]
Thread devices detect when the Leader is disconnected by observing that they are no longer receiving updates from the Leader. This timeout is set to a few minutes.

**Reply by muralidhar-bn** [reply]
Leader re-election taking few minutes is having any defined min and max value ? Can we configure timeout to enforce the Leader re-election ? Also can user enforce new leader election for required otbr device (Raspberry Pi 4 ) ?

**Reply by jwhui** [reply]
The Network ID Timeout specifies the maximum duration between receiving such updates from the Leader. The Thread Specification specifies this at 120 seconds.
https://github.com/openthread/openthread/blob/db6393251a25b9952a4907137092acf9c7f254aa/src/core/thread/mle_router.hpp#L554

OpenThread provides APIs to changes this timeout, but changing it would not conform to the Thread Specification.

The only way to "force" a new leader election is by having a device advertise a higher Leader Weight. Here is one proposal https://github.com/openthread/openthread/pull/9106/commits/4dde33ab797a352a7ea481e5d8bbf88620808682 (but was never merged).

**Reply by xiaojiang2017** [reply]
Hi @jwhui 
The leader election process can cause network unavailability, which is more noticeable in a large network as it creates new partitions and other devices reattach. Is correct?

**Reply by jwhui** [reply]
Forming a new partition can take some time and does take longer as the network size grows. Generally, the Leader should be a relatively stable device. A relevant video: https://www.youtube.com/watch?v=0WXcu_r_lvQ

**Reply by vasukanster** [reply]
@jwhui Thanks for you kind answers and support, we are new to this Thread protocol
Below is one of our test setup for testing
OTBR connects to-->Nanoleaf bulb1 connects to --> Nanoleaf bulb2 (everything is within 15meters)
Sometimes we noticed that OTBR is not receiving any service details (like SRP records info) for Nanoleaf bulbs even though devices are placed nearby.
Is the protocol/otbr side any configurations can we try to increase the detection logics so that we can enhance the protocol behavior?

**Reply by jwhui** [reply]
What firmware version is running on the Nanoleaf bulbs?

**Reply by gadekula** [reply]
![IMG_20240906_232135](https://github.com/user-attachments/assets/2be73a5f-42f2-4734-b434-601bf79094e9)
Please find the version 3.6.196

**Reply by jwhui** [reply]
The Nanoleaf bulbs should be sending SRP Update messages when they attach to the network.

Are you observing that the SRP server table is empty? For example, with the `srp server service` command?

Note that this discussion has diverged to a different topic. You may want to open a separate discussion.

**Reply by vasukanster** [reply]
Yes, we observed srp server service is empty. Yes we can have separate discussion
